lib/osf-components/addon/components/search-page/component.ts

```diff
@@ -263,7 +263,7 @@ export default class SearchPage extends Component<SearchArgs> {
             this.nextPageCursor = searchResult.nextPageCursor;
             this.prevPageCursor = searchResult.prevPageCursor;
             this.searchResults = searchResult.searchResultPage.toArray();
-            this.totalResultCount = searchResult.totalResultCount === ShareMoreThanTenThousand ? '10,000+' :
+            this.totalResultCount = searchResult.totalResultCount['@id'] === ShareMoreThanTenThousand ? '10,000+' :
```

Comment:
This bit where we access `['@id']` here feels a bit strange here. If we explicitly cast `searchResult` as an `IndexCardSearchModel` (which I feel like I should have done, as `queryRecord` just returns type of `any`) in line 248 above, this would be a type error. Should we update the `totalResultCount` definition in the `index-card-search` model to expect either a number or `{ '@id': string }` like object? Not sure if that's going to make the transform weird though 🤔